### ISSUE TITLE
[BUG] (Wayland) Compositors are called (X11) Window Managers

### ISSUE THREAD

**Issue by C0rn3j** [fastfetch-cli/fastfetch#1365]
# General description of bug:

* What happened: `WM: KWin (Wayland)`
* What should happen: `Wayland Compositor: KWin` || `Compositor: KWin (Wayland)`  || `Compositor(Wayland): Kwin`  

Window Managers are an X11 thing, while Wayland compositors are a Wayland thing.

I suggest the entry reads `WM` for X11 WMs and `Compositor` or `Wayland Compositor` for Wayland compositors.

https://wiki.archlinux.org/title/Wayland#Compositors  
vs  
https://wiki.archlinux.org/title/Window_manager#List_of_window_managers

**Comment by CarterLi** [comment]
> Window Managers are an X11 thing

It's not accurate. [The definition of Window Manager in Wikipedia](https://en.wikipedia.org/wiki/Window_manager):

> A window manager is system software that controls the placement and appearance of windows within a windowing system in a graphical user interface.

It's definitely not only an X11 thing.

What you said is [X window manager](https://en.wikipedia.org/wiki/X_window_manager). Sometimes a WM is used as a synonym of X window manager because X window manager has its own process (e.g. kwin_x11). For Wayland, there are no standalone window manager **processes**, because its function is integrated into the compositor process, but the function still exists. Wikipedia still calls them [window](https://en.wikipedia.org/wiki/Compositing_window_manager#X11_and_Wayland) [managers](https://en.wikipedia.org/wiki/Tiling_window_manager#Wayland)

TLDR: `Wayland Compositor ≈ Display Server + Window Manager`

BTW: In Windows and macOS, there are also Window Managers ([DWM](https://en.wikipedia.org/wiki/Desktop_Window_Manager) & [QC](https://en.wikipedia.org/wiki/Quartz_Compositor)), they don't use X11 or Wayland at all though.

**Comment by CarterLi** [comment]
> I suggest the entry reads WM for X11 WMs and Compositor or Wayland Compositor for Wayland compositors.

`WM` here is the name of this module; it's used in config files, command line flags and other places. A fastfetch module can only have one name (hard coded). It's can be either `WM` or `Compositor`, but not both.

**Comment by CarterLi** [comment]
There are other module names that are not accurate. The `Bios` module should have been `Bootloader` or `Firmware` (but not `UEFI`); The `Disk` module should have been `Mountpoint` or `Partition` while `Physical Disk` should be `Disk`. However, in order to keep backward compatibility, they will never be changed, probably.

**Comment by C0rn3j** [comment]
> Bios module should have been Bootloader or Firmware
> but not UEFI

As per the UEFI spec, BIOS is UEFI Class 0, so it does fall under UEFI.  
Not that I am a huge fan of that, but that would beat the current `Bios (UEFI):`, where BIOS is wrong, and isn't even capitalized.

https://en.wikipedia.org/wiki/UEFI#Classes

> However, in order to keep backward compatibility, they will never be changed, probably.

Do you have to break backwards compatibility by changing the internals too?

Or you could change the internals to support both new and old names, warn for the old names, and remove the old names at some point in the future.
